### PR TITLE
More performance optimizations

### PR DESCRIPTION
**Pull request by arkpar**
Replaced hash DB index with tree index. Hash index is not really required for prefix queries and is 10% slower. DB upgrade is not required.

### CODE REVIEW

**Comment by rphmeier** [comment]
might also be time to optimize `ordered_trie_root` and friends. these are fairly inefficient as things stand.

File: parity/main.rs [hunk 1]
```diff
@@ -487,6 +487,10 @@ fn execute_import(conf: Configuration) {
 			}
 		}
 	}
+	while !client.queue_info().is_empty() {
+		sleep(Duration::from_secs(1));
+		informant.tick(client.deref(), None);
```
Comment (rphmeier) [inline parity/main.rs]: `&*client`

**Comment by gavofyork** [comment]
will need resolving...

**Comment by rphmeier** [comment]
coveralls seems to just make numbers up. don't believe his lies

**Comment by arkpar** [comment]
rebased

File: ethcore/src/blockchain/blockchain.rs [hunk 2]
```diff
@@ -613,7 +613,7 @@ impl BlockChain {
 		let hash = block.sha3();
 		let number = header.number();
 		let parent_hash = header.parent_hash();
-		let parent_details = self.block_details(&parent_hash).expect(format!("Invalid parent hash: {:?}", parent_hash).as_ref());
+		let parent_details = self.block_details(&parent_hash).unwrap_or_else(|| panic!("Invalid parent hash: {:?}", parent_hash));
```
Comment (gavofyork) [inline ethcore/src/blockchain/blockchain.rs]: would be nice to remove these eventually.

**Pull request merged**